### ISSUE TITLE
Not detecting odd versions properly

### ISSUE THREAD

**Issue by mcg1969** [Anaconda-Platform/anaconda-project#202]
Some R packages use a mix of underscores and periods to delimit version numbers; e.g., [RJDBC](https://cran.r-project.org/web/packages/RJDBC/index.html). Conda then translates the dash to an underscore; e.g., `0.2-7.1` becomes `0.2_7.1`. Semantically, we need to treat this like `0.2.7.1`. But Anaconda Project is not matching this version number properly, and as a result, attempts to reinstall the package, and others with this idiom.

For example, this `anaconda-project.yml` file will force a reinstall of `r-rjdbc` and `r-rhive`:
```
name: R
description: A project template that contains commonly used data science packages for R.
channels: []
packages: []
platforms:
  - linux-64
  - osx-64
  - win-64
env_specs:
  anaconda50_r:
    packages:
      - r-base=3.5
      - r-essentials=3.5
      - r-sparklyr=0.7
      - r-implyr=0.2
      - r-rhive=2.0
      - r-rjdbc=0.2
      - openjdk=8.0
    channels: []
```

**Comment by mcg1969** [comment]
It should be noted that this is definitely a problem with anaconda-project; `conda` does recognize these versions properly. For instance, `conda info r-rjdbc=0.2` will include `0.2_7.1`.

**Comment by mcg1969** [comment]
@fpliger this one is having consequences in AE because it is forcing a reinstall of `r-rhive` and `r-rjdbc` in the prebaked environment.

**Comment by mcg1969** [comment]
Even though this isn't really fixed by #203, the severe consequences of it are. I will leave this open even after #203 is merged but it will be a lower priority.
